flocker/node/agents/testtools.py

```diff
@@ -117,7 +191,21 @@ def tidy_cinder_client_for_test(test_case):
     wrapped by a ``TidyCinderVolumeManager`` and register a ``test_case``
     cleanup callback to remove any volumes that are created during each test.
     """
-    client = cinder_client_from_environment()
+    client_factory = client_from_environment(client_factory=CinderClient)
+    client = client_factory(version=1)
     client.volumes = TidyCinderVolumeManager(client.volumes)
     test_case.addCleanup(client.volumes._cleanup)
     return client
+
+
+def tidy_nova_client_for_test(test_case):
+    """
+    Return a ``novaclient.v2.client.Client`` whose ``volumes`` API is a
+    wrapped by a ``TidyNovaVolumeManager`` and register a ``test_case``
+    cleanup callback to detach any volumes that are attached during each test.
+    """
+    client_factory = client_from_environment(client_factory=NovaClient)
+    client = client_factory(version=2)
+    client.volumes = TidyNovaVolumeManager(client.volumes)
+    test_case.addCleanup(client.volumes._cleanup)
+    return client
```

Comment:
Yea I guess doing attach and detach at the same time might have saved some work.